flink-core/src/main/java/org/apache/flink/api/common/operators/MailboxExecutor.java

```diff
@@ -86,6 +87,25 @@ public interface MailboxExecutor {
     /** A constant for empty args to save on object allocation. */
     Object[] EMPTY_ARGS = new Object[0];
 
+    /** Extra options to configure enqueued mails. */
+    @Experimental
+    interface MailOptions {
+        static MailOptions options() {
+            return new MailOptionsImpl();
```

Comment:
If the class would be immutable then we could use constants here to avoid objects creation/gc.

Reply:
How would you envision this to grow/develop in the future? For every combination of potential options to have constants? 🤔 

Ditto about immutability. Creating a copy of an object on every setter call would be painful. Or do you mean to have the full builder pattern - mutable public API class, that builds immutable internal class?

Reply:
> How would you envision this to grow/develop in the future? For every combination of potential options to have constants? 🤔

I'd have constants for the most common combinations.

> Ditto about immutability. Creating a copy of an object on every setter call would be painful. Or do you mean to have the full builder pattern - mutable public API class, that builds immutable internal class?

Can't we just use factory?
And maybe hide the constructor to make things more flexible - that should allow to switch from class to an interface if necessary in the future.

Maybe I'm lacking some context, but I don't envision many options here. Do you?
If we expect say 5+ options soon then yes, builder would make more sense to me.

Reply:
I've went with factory patter for now. Can you take a look?